### PR TITLE
Run integration tests in parallel

### PR DESCRIPTION
**Pull request by ppalaga**

### CODE REVIEW

**Comment by ppalaga** [comment]
I am curious whether the actions/setup-java cache also covers the snapshots built by the needed jobs. If not, the individual native jobs will fail.

**Comment by gsmet** [comment]
It's not recommended to use the cache to propagate artifacts from one job to another as the cache lookup/save can fail. It is recommended to upload/download an artifact instead. That's what we do in the Quarkus CI: the artifacts built in the Initial JDK 17 build are propagated to the other jobs.

**Comment by ppalaga** [comment]
> It's not recommended to use the cache to propagate artifacts from one job to another as the cache lookup/save can fail. 

Thanks, good to know.

> It is recommended to upload/download an artifact instead. That's what we do in the Quarkus CI: the artifacts built in the Initial JDK 17 build are propagated to the other jobs.

We do the same in CQ and QCXF. Let me do that that way here too.

**Comment by ppalaga** [comment]
14 mins - not bad.

**Comment by geoand** [comment]
Very nice improvement, thanks a lot!

@gsmet can you confirm that this is the recommended way to go?

File: integration-tests/pom.xml [hunk 1]
```diff
@@ -18,11 +18,11 @@
     <module>simple-ollama</module>
     <module>azure-openai</module>
     <module>devui</module>
-    <module>in-process-embedding-all-minilm-l6-v2-q</module>
-    <module>in-process-embedding-all-minilm-l6-v2</module>
-    <module>in-process-embedding-bge-small-en-q</module>
-    <module>in-process-embedding-bge-small-en</module>
-    <module>in-process-embedding-e5-small-v2-q</module>
-    <module>in-process-embedding-e5-small-v2</module>
+    <module>embed-all-minilm-l6-v2-q</module>
```
Comment (cescoffier) [inline integration-tests/pom.xml:21]: Why changing the module names?

**Review by cescoffier — APPROVED** [review]
I wonder why did you change the module names?

File: integration-tests/pom.xml (hunk 1 above)
Comment (ppalaga) [reply integration-tests/pom.xml:21]: Just to match the common conventions. Is there a special reason to keep these special?

**Comment by ppalaga** [comment]
Any feedback on this @gsmet? I am open to change anything.

... and I have the BOM PR waiting in the queue.

File: integration-tests/pom.xml (hunk 1 above)
Comment (geoand) [reply integration-tests/pom.xml:21]: We used the names that are upsteam langchain4j uses, but since this is just about ITs, I don't think it matters much

Comment (cescoffier) [reply integration-tests/pom.xml:21]: yes, I think it's fine in this case.

File: .github/workflows/build-pull-request.yml [hunk 2]
```diff
@@ -20,29 +20,119 @@ defaults:
...
+        run: |
+          tar -czf ${{ runner.temp }}/maven-repo.tgz -C ~ .m2/repository
+          # Avoid caching our own artifacts
+          rm -Rf ~/.m2/repository/io/quarkiverse/langchain4j
+      - name: Persist the Maven repo
+        uses: actions/upload-artifact@v4
+        with:
+          name: maven-repo
+          path: ${{ runner.temp }}/maven-repo.tgz
+          retention-days: 1
```
Comment (gsmet) [inline .github/workflows/build-pull-request.yml:61]: I would advise to keep them for 5 days. I used to be very aggressive in Quarkus and then it became very unpractical when you want to rerun a specific subjob after a week-end.

File: .github/workflows/build-pull-request.yml [hunk 3]
```diff
@@ -20,29 +20,119 @@ defaults:
...
+    strategy:
+      fail-fast: false
+      matrix: ${{ fromJson(needs.build-and-test-jvm.outputs.matrix) }}
+    runs-on: ubuntu-latest
+
+    steps:
+
+      - name: Prepare git
+        run: git config --global core.autocrlf false
+        if: startsWith(matrix.os, 'windows')
```
Comment (gsmet) [inline .github/workflows/build-pull-request.yml:116]: This looks odd given you run all your matrix on Ubuntu?

**Review by gsmet — COMMENTED** [review]
Looks good in general.

I added one advice and one comment.

File: .github/workflows/build-pull-request.yml (hunk 3 above)
Comment (ppalaga) [reply .github/workflows/build-pull-request.yml]: I copied it from the old main job without too much thinking. Thanks for the point, removed.

File: .github/workflows/build-pull-request.yml (hunk 2 above)
Comment (ppalaga) [reply .github/workflows/build-pull-request.yml]: +1, increased to 5 days

**Comment by ppalaga** [comment]
491a318:
* Rebased
* Addressed the suggestions from @gsmet

**Comment by ppalaga** [comment]
Anybody has an idea what the deadlock in azure-openai test means? I have not seen anything like that before.

**Comment by geoand** [comment]
Unfortunately that is a bug we have encountered in GraalVM since day 1 of this extension. 
@zakkak said he will look into it when time permits, but for now there is nothing we can do.

**Pull request merged**